apps/factory/src/components/create_template.tsx

```diff
@@ -139,14 +240,29 @@ export function CreateTemplate({
           <Button size="small" onClick={handleClose} variant="outlined">
             Cancel
           </Button>
-          <Button
-            size="small"
-            onClick={handleSubmit(onFormSubmit)}
-            variant="outlined"
-            color="primary"
-          >
-            Confirm
-          </Button>
+          {ptId ? (
+            <>
+              <Button
+                size="small"
+                onClick={handleSubmit(updateTemplate)}
+                variant="outlined"
+                color="primary"
+              >
+                Confirm
+              </Button>
+            </>
+          ) : (
+            <>
+              <Button
+                size="small"
+                onClick={handleSubmit(onFormSubmit)}
+                variant="outlined"
+                color="primary"
+              >
+                Confirm
+              </Button>
+            </>
+          )}
```

Comment:
this code is exactly same apart from onclick, please make it conditional onclick